manager/src/main/java/com/redhat/developer/manager/api/user/ConnectorsAPI.java

```diff
@@ -0,0 +1,43 @@
+package com.redhat.developer.manager.api.user;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import com.redhat.developer.manager.ConnectorsService;
+import com.redhat.developer.manager.CustomerIdResolver;
+import com.redhat.developer.manager.models.Connector;
+import com.redhat.developer.manager.requests.ConnectorRequest;
+
+@Path("/connectors")
+public class ConnectorsAPI {
+
+    @Inject
+    CustomerIdResolver customerIdResolver;
+
+    @Inject
+    ConnectorsService connectorsService;
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getConnectors() {
```

Comment:
We have the requirement to follow the guidelines here https://gitlab.cee.redhat.com/service/api-guidelines for any API we expose to our customers. So at some point I think we're going to want to create a `ConnectorsResponse` object that includes things like the HATEOS links, versions etc.

So I'd suggest we do that now. Take a look at the guidelines I've linked ^ and also how we did it in DaaS here: https://github.com/kiegroup/baaas-decision-fleet-manager/tree/main/baaas-decision-fleet-manager-api/src/main/java/org/kie/baaas/dfm/api/decisions

You can then see how we mapped between API objects and internal domain model in DaaS here: https://github.com/kiegroup/baaas-decision-fleet-manager/blob/main/baaas-decision-fleet-manager/src/main/java/org/kie/baaas/dfm/app/controller/modelmappers/DecisionMapper.java

Reply:
> We have the requirement to follow the guidelines here https://gitlab.cee.redhat.com/service/api-guidelines for any API we expose to our customers. So at some point I think we're going to want to create a `ConnectorsResponse` object that includes things like the HATEOS links, versions etc.
> 
> So I'd suggest we do that now. 

So let me summarize what we have to implement in this PR to follow the guidelines:
1) authentication 
2) Representation (kind, id, href) with a reference to the error description - something we did not have in baaas
3) pagination 
4) search 
5) Documentation with OpenAPI

From my side it is not a big deal to include those in this PR, but, If we want to do this in this PR, I suggest we edit the MGDOBR-1 ticket so to include all of those tasks. This PR will become bigger and I need probably some more days to do the job. 
Let me know how you prefer to proceed @robpblake

Reply:
I've added the fields according to the specification. the `href` field is present in the definition but not populated yet because I'll implement the GET /connectors/{id} endpoint in the next pull request (this one is already big)

Reply:
Thanks @r00ta. Works for me 👍